metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/data_resolver.py

```diff
@@ -0,0 +1,643 @@
+import logging
+from abc import ABC, abstractmethod
+from time import sleep
+from typing import Any, List, Optional
+
+import msal
+import requests
+
+from datahub.configuration.common import ConfigurationError
+from datahub.ingestion.source.powerbi.config import Constant
+from datahub.ingestion.source.powerbi.rest_api_wrapper.data_classes import (
+    Dashboard,
+    Page,
+    PowerBIDataset,
+    Report,
+    Tile,
+    User,
+    Workspace,
+    new_powerbi_dataset,
+)
+
+# Logger instance
+logger = logging.getLogger(__name__)
+
+
+def is_permission_error(e: requests.exceptions.HTTPError) -> bool:
+    return e.response.status_code == 401 or e.response.status_code == 403
```

Comment:
Des this always have response object, right?

Reply:
yes